Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -73,6 +73,20 @@ class DirectNavigator {
         // Initialize the surface sequence
         state.navigation.navSurfaces = navSurfaces;
         state.navigation.navSurfaceIter = state.navigation.navSurfaces.begin();
+
+        // In case the start surface is in the list of nav surfaces
+        // we need to correct the iterator to point to the next surface
+        // in the vector
+        if (state.navigation.startSurface) {
```

Comment:
I think the start surface is/should be excluded from the navigation stream :+1:

Reply:
exactly. But i don't think it is always assured the first surface is the start surface, right? thus this check.

Reply:
yeah I think in principle someone might start at a different surface in the stream. so it is good to support that but as you said the interface is unclear